### PR TITLE
[FEATURE] Improve Trino Support

### PR DESCRIPTION
**Pull request by aezomz**
Please annotate your PR title to describe what the PR does, then give a brief bulleted description of your PR below. PR titles should begin with [BUGFIX], [FEATURE],  [DOCS], or [MAINTENANCE]. If a new feature introduces breaking changes for the Great Expectations API or configuration files, please also add [BREAKING]. You can read about the tags in our [contributor checklist](https://docs.greatexpectations.io/docs/contributing/contributing_checklist).

Changes proposed in this pull request:
- Trino to create view for profiling
- Included handling for Trino dialect which is similar to Athena
- Added exceptional handling for `stddev_samp` which return blank when sampling some values
- Added trino.sqlalchemy.datatype to retrieve potential data types

After submitting your PR, CI checks will run and @cla-bot will check for your CLA signature.

For a PR with nontrivial changes, we review with both design-centric and code-centric lenses.

In a design review, we aim to ensure that the PR is consistent with our relationship to the open source community, with our software architecture and abstractions, and with our users' needs and expectations. That review often starts well before a PR, for example in github issues or slack, so please link to relevant conversations in notes below to help reviewers understand and approve your PR more quickly (e.g. `closes #123`).

Previous Design Review notes:
-
-
-

### Definition of Done
Please delete options that are not relevant.

- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added [unit tests](https://docs.greatexpectations.io/docs/contributing/contributing_test#writing-unit-and-integration-tests) where applicable and made sure that new and existing tests are passing.
- [x] I have run any local integration tests and made sure that nothing is broken.

Thank you for submitting!

### CODE REVIEW

**Comment by aezomz** [comment]
@kenwade4 I think most of the tests should pass now. I have rebased to the latest branch too. 
There is no datetime in Trino which cause the test to fail. Is it possible to exclude datetime tests?

Is there a way to change the CLA email? I created a new PR now. But it still doesn't pull the correct git config. 

I am available on slack as well.

**Comment by kyleaton** [comment]
@cla-bot check

**Comment by aezomz** [comment]
@kenwade4 I managed to pass all the db_integration test. 

Do I need to fix the partner_integrations too? I am not sure if i have edited anything that affects it.

**Comment by kenwade4** [comment]
@aezomz I think that might be a CI environment issue on our end. I'm going to test building the Expectation gallery locally from this PR and I think this should be good to go!

**Comment by aezomz** [comment]
Thanks @kenwade4, do let me know if you need any help.

**Comment by kenwade4** [comment]
Hi @aezomz. When building the Expectation gallery locally on this branch, none of the Expectations are able to successfully run their example tests against Trino. To attempt this yourself, you can start a Trino test container (`cd assets/docker/trino; docker-compose up`) in one terminal, then run the `build_gallery.py` script in another (`source venv/bin/activate; cd assets/scripts; python ./build_gallery.py --no-contrib`). There will be a lot of debug output shown and at the end you will see the diagnostic checklist output for each Expectation. To speed things up, you can run the script with some specific Expectation names passed in, instead of all Expectations (`python ./build_gallery.py expect_column_max_to_be_between expect_column_min_to_be_between`)

The checklist output at the end should be something like the following

```
----------------
expect_column_max_to_be_between (core)
Completeness checklist for ExpectColumnMaxToBeBetween (PRODUCTION):
 ✔ Has a valid library_metadata object
 ✔ Has a docstring, including a one-line short description
    ✔ "Expect the column max to be between an min and max value"
 ✔ Has at least one positive and negative example case, and all test cases pass
 ✔ Has core logic and passes tests on at least one Execution Engine
    ✔ All 6 tests for pandas are passing
    ✔ All 6 tests for sqlite are passing
    ✔ All 6 tests for trino are passing
 ✔ Passes all linting checks
 ✔ Has basic input validation and type checking
      Using default validate_configuration from template
 ✔ Has both statement Renderers: prescriptive and diagnostic
 ✔ Has core logic that passes tests for all applicable Execution Engines and SQL dialects
    ✔ All 6 tests for pandas are passing
    ✔ All 6 tests for sqlite are passing
    ✔ All 6 tests for trino are passing
 ✔ Has a full suite of tests, as determined by a code owner
 ✔ Has passed a manual review by a code owner for code standards and style guides



----------------
expect_column_min_to_be_between (core)
Completeness checklist for ExpectColumnMinToBeBetween (PRODUCTION):
 ✔ Has a valid library_metadata object
 ✔ Has a docstring, including a one-line short description
    ✔ "Expect the column minimum to be between an min and max value"
 ✔ Has at least one positive and negative example case, and all test cases pass
 ✔ Has core logic and passes tests on at least one Execution Engine
    ✔ All 8 tests for pandas are passing
    ✔ All 8 tests for sqlite are passing
    ✔ All 8 tests for trino are passing
 ✔ Passes all linting checks
 ✔ Has basic input validation and type checking
      Using default validate_configuration from template
 ✔ Has both statement Renderers: prescriptive and diagnostic
 ✔ Has core logic that passes tests for all applicable Execution Engines and SQL dialects
    ✔ All 8 tests for pandas are passing
    ✔ All 8 tests for sqlite are passing
    ✔ All 8 tests for trino are passing
 ✔ Has a full suite of tests, as determined by a code owner
 ✔ Has passed a manual review by a code owner for code standards and style guides
```

If you run on the `develop` branch, you will see this. This PR cannot be accepted until this is resolved.

File: great_expectations/self_check/util.py [hunk 1]
```diff
@@ -673,14 +673,15 @@ def get_dataset(
...
-            table_name = generate_test_table_name()
+            table_name = generate_test_table_name().lower()
 
         df.to_sql(
             name=table_name,
             con=engine,
             index=False,
             dtype=sql_dtypes,
             if_exists="replace",
+            method="multi",
```
Comment (kenwade4) [inline great_expectations/self_check/util.py:684]: Please consider this for defining the table_name and value for method kwarg https://github.com/great-expectations/great_expectations/blob/939be373fb22e10c809aa488e7f1a3521327ed65/great_expectations/self_check/util.py#L1176-L1191

File: docs/changelog.md [hunk 2]
```diff
@@ -57,6 +57,7 @@ title: Changelog
 * [MAINTENANCE] Fix docstring on expectation (#5204) (thanks @andyjessen)
 * [MAINTENANCE] Improve NaN handling in numeric ParameterBuilder implementations (#5226)
 * [MAINTENANCE] Update type hint and docstring linter thresholds (#5228)
+* [FEATURE] Improve Trino support and enable test (#5261)
```
Comment (kenwade4) [inline docs/changelog.md]: Please remove this line. After this PR is approved and merged, the changelog will be automatically updated on the next release.

**Comment by aezomz** [comment]
```
----------------
expect_column_min_to_be_between (core)
Completeness checklist for ExpectColumnMinToBeBetween (EXPERIMENTAL):
 ✔ Has a valid library_metadata object
 ✔ Has a docstring, including a one-line short description
    ✔ "Expect the column minimum to be between an min and max value"
   Has at least one positive and negative example case, and all test cases pass
   Has core logic and passes tests on at least one Execution Engine
      There are no test results
 ✔ Passes all linting checks
   Has basic input validation and type checking
      No example found to get kwargs for ExpectationConfiguration
 ✔ Has both statement Renderers: prescriptive and diagnostic
   Has core logic that passes tests for all applicable Execution Engines and SQL dialects
      There are no test results
 ✔ Has a full suite of tests, as determined by a code owner
 ✔ Has passed a manual review by a code owner for code standards and style guides

```

@kenwade4  i am running this on latest develop branch and follow your steps. 

I am not able to get any test running. Did u updated with your own changes?

**Comment by aezomz** [comment]
@kenwade4 I merged the latest dev. and the gallery tests are running now.

**Comment by kenwade4** [comment]
@aezomz I just pulled this branch and I don't see any trino gallery tests when I run the script

**Comment by aezomz** [comment]
@kenwade4  Weird, it should be running.
I get result like below 
```
----------------
expect_column_values_to_be_in_set (core)
Completeness checklist for ExpectColumnValuesToBeInSet (EXPERIMENTAL):
 ✔ Has a valid library_metadata object
 ✔ Has a docstring, including a one-line short description
    ✔ "Expect each column value to be in a given set."
   Has at least one positive and negative example case, and all test cases pass
 ✔ Has core logic and passes tests on at least one Execution Engine
    ✔ All 15 tests for pandas are passing
    ✔ All 10 tests for sqlite are passing
    ✔ All 10 tests for trino are passing
 ✔ Passes all linting checks
 ✔ Has basic input validation and type checking
    ✔ Custom 'assert' statements in validate_configuration
 ✔ Has both statement Renderers: prescriptive and diagnostic
   Has core logic that passes tests for all applicable Execution Engines and SQL dialects
    ✔ All 15 tests for pandas are passing
    ✔ All 10 tests for sqlite are passing
    ✔ All 10 tests for trino are passing
      Only 11 / 12 tests for spark are passing
        - Failing: basic_negative_test_case_datetime_set
 ✔ Has a full suite of tests, as determined by a code owner
 ✔ Has passed a manual review by a code owner for code standards and style guides
```

**Comment by aezomz** [comment]
Maybe try to re install in a new venv? thats what i did to make the test appearing with develop branch then i switch to this branch to run the gallery... 
Is it missing Trino package?

```
python3 -m venv ge_dev

source ge_dev/bin/activate

pip install --upgrade pip==21.3.1

pip install -r requirements.txt -r requirements-dev-lite.txt -r requirements-dev-trino.txt -c constraints-dev.txt -e .
```

**Comment by aezomz** [comment]
I merged the latest dev again. 
I removed my venv and re-install from scratch with this branch. 

I am able to finish running the gallery `python assets/scripts/build_gallery.py --no-contrib`
With no failing Trino tests.

File: tests/conftest.py [hunk 3]
```diff
@@ -187,7 +187,7 @@ def build_test_backends_list(metafunc):
     return [
         (backend_name_class_name_map.get(backend_name) or backend_name)
         for backend_name in test_backend_names
-        if backend_name != "trino"  # <--------- maybe remove that
+        # if backend_name != "trino"  # <--------- maybe remove that
```
Comment (kenwade4) [inline tests/conftest.py]: Can you please delete this line outright

File: tests/test_definitions/column_map_expectations/expect_column_values_to_be_in_set.json [hunk 4]
```diff
@@ -226,7 +226,7 @@
     "tests" : [
       {
         "title": "basic_positive_test_case_datetime_set",
-        "suppress_test_for": ["sqlite"],
+        "suppress_test_for": ["sqlite", "trino"],
```
Comment (kenwade4) [inline tests/test_definitions/column_map_expectations/expect_column_values_to_be_in_set.json]: Don't suppress this datetime test for Trino, as we will be working with their engineers to get this to work

File: tests/test_definitions/column_map_expectations/expect_column_values_to_be_in_set.json [hunk 5]
```diff
@@ -239,7 +239,7 @@
     },
     {
       "title": "basic_negative_test_case_datetime_set",
-      "suppress_test_for": ["sqlite", "bigquery"],
+      "suppress_test_for": ["sqlite", "bigquery", "trino"],
```
Comment (kenwade4) [inline tests/test_definitions/column_map_expectations/expect_column_values_to_be_in_set.json]: Don't suppress this datetime test for Trino, as we will be working with their engineers to get this to work

**Review by kenwade4 — CHANGES_REQUESTED** [review]
@aezomz overall fantastic work and I appreciate your diligence! And a fresh venv got around the local build_gallery issue. Just a few line deletions and it will be approved and merged. Thanks again!

**Review by kenwade4 — APPROVED** [review]

**Comment by aezomz** [comment]
Thanks @kenwade4 for reviewing this PR. 👍 Have a good day!

**Pull request merged**